src/server/game/DataStores/DBCStores.cpp

```diff
@@ -717,10 +724,222 @@ void LoadDBCStores(const std::string& dataPath)
         exit(1);
     }
 
+    LoadM2Cameras(dataPath);
+
     TC_LOG_INFO("server.loading", ">> Initialized %d data stores in %u ms", DBCFileCount, GetMSTimeDiffToNow(oldMSTime));
 
 }
 
+// Convert the geomoetry from a spline value, to an actual WoW XYZ
+M2Vector3 TranslateLocation(M2Vector4 const* DBCPosition, M2Vector3 const* basePosition, M2Vector3 const* splineVector)
+{
+    M2Vector3 work;
+    float x = basePosition->x + splineVector->x;
+    float y = basePosition->y + splineVector->y;
+    float z = basePosition->z + splineVector->z;
+    float const distance = sqrt((x * x) + (y * y));
+    float angle = std::atan2(x, y) - DBCPosition->o;
+
+    if (angle < 0)
+        angle += 2 * float(M_PI);
+
+    work.x = DBCPosition->x + (distance * sin(angle));
+    work.y = DBCPosition->y + (distance * cos(angle));
+    work.z = DBCPosition->z + z;
+    return work;
+}
+
+// Number of cameras not used. Multiple cameras never used in 3.3.5
+void readCamera(M2Camera const* cam, uint32 /*numCams*/, M2Header const* header, CinematicCameraEntry const* dbcentry)
```

Comment:
You should just do all of this preprocessing in map extractor and only write out stuff you will use in worldserver (and use G3D types for vectors/aabox instead of defining your own)